FirebasePerformance/Tests/Unit/FPRSessionManagerTest.m

```diff
@@ -34,45 +36,28 @@ - (void)testInstanceCreation {
   XCTAssertEqual(instance, [FPRSessionManager sharedInstance]);
 }
 
-/** Validate that valid sessionId always exists. */
-- (void)testSessionIdExistance {
-  FPRSessionManager *instance = [FPRSessionManager sharedInstance];
-  [instance startTrackingAppStateChanges];
-
-  NSNotificationCenter *notificationCenter = [NSNotificationCenter defaultCenter];
-  [notificationCenter postNotificationName:UIApplicationWillEnterForegroundNotification
-                                    object:[UIApplication sharedApplication]];
-
-  XCTAssertNotNil(instance.sessionDetails.sessionId);
-
-  NSString *lowercaseSessionId = [instance.sessionDetails.sessionId lowercaseString];
-  XCTAssertEqualObjects(lowercaseSessionId, instance.sessionDetails.sessionId);
-}
-
-/** Validate that sessionId does not change when calling renew method immediately. */
-- (void)testSessionIdNotGettingRenewed {
+/** Validate that gauge collection does not change when calling renew method immediately. */
+- (void)testGaugeDoesNotStopBeforeMaxDuration {
```

Comment:
Do we have tests for stopping gauge collection after the session has been running for too long? You can try this by overriding the max duration config and see if we are stopping to collect gauge events after that.

Reply:
I am trying to write this right now, but it's super hard with all the singletons (`[FPRConfigurations sharedInstance]` is directly called in code, no way to pass in a fake) basically blocking me from injecting dependencies. Going with the overriding max duration in config approach, minimum value cannot be 0 minutes because it would be auto-overwritten to default 240 minutes. 1 minute would be too long to wait. This is why I prefer no-singletons, and passing in `Date` as current time, but I can't pass in time because `FPRSessionDetails.sessionLengthInMinutes` is a property.

Reply:
Done, by adding a `(NSDate *)now` param to `stopGaugesIfRunningTooLong`